### DISCUSSION TITLE
uiSlider Issue [SOLVED]

### DISCUSSION THREAD

**Discussion by spiffo-smith** [RomanKubiak/ctrlr#474] [General]
I've never used uiSliders before, I was just doing some testing and noticed this odd behaviour!

I'm prepared to feel really dumb but if I open a new panel and create a uiSlider (default settings, I make no changes) why does it NOT change it's value? All other Slider types work as expected and change their values!

**Comment by Godlike-Productions** [comment]
You need to make sure you have an image that is set up for the uiImage.  It only works with vertically arrayed images, and you will need to know the frame size for each image in the array. jKnobman makes images that work, if you set it to output vertical arrays.

**Reply by spiffo-smith** [reply]
Isn't that for a uiImageSlider?

I'm talking about Right Click on Panel, Add Component - Sliders - uiSlider

When I add one, it doesn't change it's value, all other Buttons and Sliders work as expected, it's very strange, I've never noticed before because I've not used them on any panels I've worked on. I must be doing something stupid?

**Reply by DRBAUS** [reply]
It's weird indeed.

What version of Ctrlr are you working with atm?

Your DX-11 panel looks great. I borrowed some of those knobs and layout for the panels that I am still working on. Thanks for answering my question about how to import images for uiImageSlider on your YT chanel a while ago. Good to see you here too!

**Reply by spiffo-smith** [reply]
Yes hello DrBaus!
 I’m still using V5.3.201 the stable one. It’s bizarre I was doing some testing and found every time I added a uiSlider it just won’t change its value, can anyone confirm or tell me what stupid thing I’m doing wrong?

**Reply by dnaldoog** [reply]
Hi @spiffo

Can you post a simple panel with said problem?

**Reply by damiensellier** [reply]
Are you using ctrlr in panel mode (Cmd+E) ?  Sometimes Ctrlr doesn't send sysex messages if not in panel mode.

**Reply by spiffo-smith** [reply]
[uiSlider Problem.zip](https://github.com/RomanKubiak/ctrlr/files/9090084/uiSlider.Problem.zip)

Panel attached.

All I've done is add a uiSlider and a uiFixedSlider. I've given the uiFixedSlider some slider contents and set both to output MIDI CC 0. Is there something else that needs to be done with a uiSlider? It defaults to a 0-127 value range so I didn't think there was anything else that needed doing with it?

**Reply by damiensellier** [reply]
![ezgif-3-565d558d16](https://user-images.githubusercontent.com/55164385/178443728-c1361153-5551-4a29-a44d-819605fc2199.gif)
Hi mate i just tried and the panel is sending midi with both sliders. 
I tried on OSX 10.15.7 with ctrlr 5.3.198
I just had to select my midi i/o settings
I could not reproduce the bug.
Not sure if it's on purpose but  uiFixedSlider value updates from uiSlider's one.

**Reply by spiffo-smith** [reply]
Thanks for testing. OK, I think I see the slight bug.

Make sure the Panel Lock option is ticked and then it does NOT work in Editing Mode but it DOES work in normal Panel Mode!

**Reply by damiensellier** [reply]
Can you please mark the discussion as solved? thanks ;)

**Reply by spiffo-smith** [reply]
Done, it appears to be a quirk that only affects uiSlider and only when simultaneously in Panel Lock + Editing Mode,
